wolfcrypt/src/asn.c

```diff
@@ -6028,14 +6030,32 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
 
         if (verify != NO_VERIFY && type != CA_TYPE && type != TRUSTED_PEER_TYPE) {
             cert->ca = NULL;
-        #ifndef NO_SKID
+    #ifndef NO_SKID
+        /* Note: This is not a secure solution and provides a workaround for
+            situations where only the intermediate CA is available for a peer
+            cert and the OCSP request for it is signed by the root CA. */
```

Comment:
Would any self-signed cert in any case also be "validated" here?  Let's document the most general case as well as this specific one.